pkg/build/builder/common.go

```diff
@@ -81,3 +86,87 @@ func updateBuildRevision(c client.BuildInterface, build *api.Build, sourceInfo *
 		glog.Warningf("An error occurred saving build revision: %v", err)
 	}
 }
+
+func runPostCommitHook(dockerClient DockerClient, build *api.Build, strategyName, imageID string) error {
+	glog.Infof("Running post commit hook with image %s ...", imageID)
+	containerName := fmt.Sprintf("openshift_%s-build_%s_%s_post-commit_%08x", strategyName, build.Name, build.Namespace, rand.Uint32())
+	return runScriptInContainer(dockerClient, build.Spec.PostCommit.RunBash, imageID, containerName)
```

Comment:
@bparees @mfojtik @smarterclayton please help me sanity checking here.

Is the only reason **not** to run the post commit script as a POD the fact that we need to reference a specific image that only exists in the node where it has been built?

This is one more difference between Build Post Commit and Deployment Lifecycle Hooks.